test/tests/jobs/RetryJobTest.cpp

```diff
@@ -334,6 +336,68 @@ struct RetryJobTest : tpunit::TestFixture {
         ASSERT_EQUAL(difftime(nextRunTime, createdTime), 3600);
     }
 
+    // Retry a job with a repeat that uses the custom "START OF HOUR" modifier
+    void hasRepeatStartOfHour() {
+        uint64_t now = STimeNow();
+
+        // Create the job
+        SData command("CreateJob");
+        command["name"] = "job";
+        command["repeat"] = "SCHEDULED, START OF DAY, +5 MINUTES, START OF HOUR";
+        command["nextRun"] = now;
```

Comment:
This needs to be `firstRun` doesn't it?

Reply:
I don't think so. The tests fail if I make it `firstRun`, and looking at the code I posted above, I think `nextRun` is correct if repeat includes `SCHEDULED`. What am I missing?

Reply:
Where is `nextRun` used in here?
https://github.com/Expensify/Bedrock/blob/main/plugins/Jobs.cpp#L376-L633

I don't see it used anywhere. Does it do anything? What if you leave it out entirely?

Reply:
You're right, `nextRun` doesn't seem to be necessary, but the test definitely fail if I set `firstRun`:

```
   exception #1 from RetryJobTest::hasRepeatStartOfHour with cause: Expected 200 OK, but got '404 No job found'. Initially thrown from: Jobs.cpp:743
❌ !FAILED! ❌ RetryJobTest::hasRepeatStartOfHour
   exception #1 from RetryJobTest::hasRepeatStartOfHourNotLast with cause: Expected 200 OK, but got '404 No job found'. Initially thrown from: Jobs.cpp:743
❌ !FAILED! ❌ RetryJobTest::hasRepeatStartOfHourNotLast
```

Gotta figure out why that is

Reply:
Ah, figured out what the problem is. I am creating `now` as a `uint64_t` but the code expects `firstRun` to be a date time string. So passing a `uint64_t` as `firstRun` was causing the job to fail to get created, which in turn caused `GetJob` to return an error. Fix incoming.